### PR TITLE
checkpoint_management.py - custom checkpoint prefix

### PR DESCRIPTION
**Pull request by ohadlights**

### CODE REVIEW

File: tensorflow/python/training/checkpoint_management.py
Comment (allenlavoie) [inline tensorflow/python/training/checkpoint_management.py]: Why None? Seems slightly clearer to just default the argument to "ckpt".

And could you add a short blurb to the Args: section of the docstring?

**Review by allenlavoie — COMMENTED** [review]
Sounds like a reasonable option to me. This will need an API review. Could you regenerate the API goldens? The readme in this directory has instructions: https://github.com/tensorflow/tensorflow/tree/master/tensorflow/tools/api/tests

File: tensorflow/python/training/checkpoint_management.py
Comment (ohadlights) [reply tensorflow/python/training/checkpoint_management.py]: Changed the default value to "ckpt" and added a desctiotion for the parameter.
Working on the API goldens...

**Comment by ohadlights** [comment]
I updated the relevant API's. hope it was done correctly.

These are the results of the test:

root@comp:/mnt/dev/tensorflow# bazel-bin/tensorflow/tools/api/tests/api_compatibility_test --update_goldens True
Limited tf.compat.v2.summary API due to missing TensorBoard installation
Running tests under Python 2.7.12: /usr/bin/python
[ RUN      ] ApiCompatibilityTest.testAPIBackwardsCompatibility
W0319 23:55:09.072984 140072935352064 deprecation.py:323] From /mnt/dev/tensorflow/bazel-bin/tensorflow/tools/api/tests/api_compatibility_test.runfiles/org_tensorflow/tensorflow/python/util/decorator_utils.py:145: VARIABLES (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.GraphKeys.GLOBAL_VARIABLES` instead.
E0319 23:55:12.594662 140072935352064 api_compatibility_test.py:246] TensorFlow API backwards compatibility test
This test ensures all changes to the public API of TensorFlow are intended.

If this test fails, it means a change has been made to the public API. Backwards
incompatible changes are not allowed. You can run the test as follows to update
test goldens and package them with your change.

    $ bazel build tensorflow/tools/api/tests:api_compatibility_test
    $ bazel-bin/tensorflow/tools/api/tests/api_compatibility_test \
          --update_goldens True

You will need an API approval to make changes to the public TensorFlow API. This
includes additions to the API.

E0319 23:55:12.595136 140072935352064 api_compatibility_test.py:247] 1 differences found between API and golden.
Issue 1 :
  path: "tensorflow.train.CheckpointManager"
  tf_class {
    is_instance: "<class \'tensorflow.python.training.checkpoint_management.CheckpointManager\'>"
    is_instance: "<type \'object\'>"
    member {
      name: "checkpoints"
      mtype: "<type \'property\'>"
    }
    member {
      name: "latest_checkpoint"
      mtype: "<type \'property\'>"
    }
    member_method {
      name: "__init__"
-     argspec: "args=[\'self\', \'checkpoint\', \'directory\', \'max_to_keep\', \'keep_checkpoint_every_n_hours\'], varargs=None, keywords=None, defaults=[\'None\'], "
+     argspec: "args=[\'self\', \'checkpoint\', \'directory\', \'max_to_keep\', \'keep_checkpoint_every_n_hours\', \'checkpoint_name\'], varargs=None, keywords=None, defaults=[\'None\', \'ckpt\'], "
?                                                                                                                +++++++++++++++++++++                                                  ++++++++++
    }
    member_method {
      name: "save"
      argspec: "args=[\'self\', \'checkpoint_number\'], varargs=None, keywords=None, defaults=[\'None\'], "
    }
  }

W0319 23:55:12.595268 140072935352064 api_compatibility_test.py:254] Golden file update requested!
All test failures have been skipped, see the logs for detected diffs.
This test is now going to write new golden files.
Make sure to package the updates together with your change.

You will need an explicit API approval. This may take longer than a normal
review.

[       OK ] ApiCompatibilityTest.testAPIBackwardsCompatibility
[ RUN      ] ApiCompatibilityTest.testAPIBackwardsCompatibilityV1
I0319 23:55:15.501441 140072935352064 api_compatibility_test.py:273] No differences found between API and golden.
[       OK ] ApiCompatibilityTest.testAPIBackwardsCompatibilityV1
[ RUN      ] ApiCompatibilityTest.testAPIBackwardsCompatibilityV2
E0319 23:55:18.709964 140072935352064 api_compatibility_test.py:246] TensorFlow API backwards compatibility test
This test ensures all changes to the public API of TensorFlow are intended.

If this test fails, it means a change has been made to the public API. Backwards
incompatible changes are not allowed. You can run the test as follows to update
test goldens and package them with your change.

    $ bazel build tensorflow/tools/api/tests:api_compatibility_test
    $ bazel-bin/tensorflow/tools/api/tests/api_compatibility_test \
          --update_goldens True

You will need an API approval to make changes to the public TensorFlow API. This
includes additions to the API.

E0319 23:55:18.710264 140072935352064 api_compatibility_test.py:247] 1 differences found between API and golden.
Issue 1 :
  path: "tensorflow.train.CheckpointManager"
  tf_class {
    is_instance: "<class \'tensorflow.python.training.checkpoint_management.CheckpointManager\'>"
    is_instance: "<type \'object\'>"
    member {
      name: "checkpoints"
      mtype: "<type \'property\'>"
    }
    member {
      name: "latest_checkpoint"
      mtype: "<type \'property\'>"
    }
    member_method {
      name: "__init__"
-     argspec: "args=[\'self\', \'checkpoint\', \'directory\', \'max_to_keep\', \'keep_checkpoint_every_n_hours\'], varargs=None, keywords=None, defaults=[\'None\'], "
+     argspec: "args=[\'self\', \'checkpoint\', \'directory\', \'max_to_keep\', \'keep_checkpoint_every_n_hours\', \'checkpoint_name\'], varargs=None, keywords=None, defaults=[\'None\', \'ckpt\'], "
?                                                                                                                +++++++++++++++++++++                                                  ++++++++++
    }
    member_method {
      name: "save"
      argspec: "args=[\'self\', \'checkpoint_number\'], varargs=None, keywords=None, defaults=[\'None\'], "
    }
  }

W0319 23:55:18.710416 140072935352064 api_compatibility_test.py:254] Golden file update requested!
All test failures have been skipped, see the logs for detected diffs.
This test is now going to write new golden files.
Make sure to package the updates together with your change.

You will need an explicit API approval. This may take longer than a normal
review.

[       OK ] ApiCompatibilityTest.testAPIBackwardsCompatibilityV2
[ RUN      ] ApiCompatibilityTest.testNoSubclassOfMessage
[       OK ] ApiCompatibilityTest.testNoSubclassOfMessage
[ RUN      ] ApiCompatibilityTest.testNoSubclassOfMessageV1
[       OK ] ApiCompatibilityTest.testNoSubclassOfMessageV1
[ RUN      ] ApiCompatibilityTest.testNoSubclassOfMessageV2
[       OK ] ApiCompatibilityTest.testNoSubclassOfMessageV2
[ RUN      ] ApiCompatibilityTest.test_session
[  SKIPPED ] ApiCompatibilityTest.test_session
----------------------------------------------------------------------
Ran 7 tests in 10.638s

OK (skipped=1)

File: tensorflow/python/training/checkpoint_management.py
Comment (allenlavoie) [inline tensorflow/python/training/checkpoint_management.py]: Nit: could you wrap the line to 80 characters?

**Review by allenlavoie — DISMISSED** [review]

File: tensorflow/python/training/checkpoint_management.py
Comment (ohadlights) [reply tensorflow/python/training/checkpoint_management.py]: Gave each parameter a line of its own. Looks better and seems to be more aligned with the conventions.

**Review by allenlavoie — DISMISSED** [review]

**Review by allenlavoie — APPROVED** [review]

**Comment by ohadlights** [comment]
Is there something i need to do? Or the tests just have a long queue?

**Comment by allenlavoie** [comment]
There's nothing you need to do. It's just waiting for an API review.

**Pull request merged**
